File: main/QuickSortTest.java
import static org.junit.Assert.*;

class QuickSortTest {

	public void test_a_normal_array() {
		int[] array = {3, 2, -5, 1, 0};
		int[] sortedArr = {-5, 0, 1, 2, 3};
		QuickSort.quicksort(array, 0, array.length-1);
		assertArrayEquals(sortedArr, array);
	}
	
	public void test_an_empty_array() {
		int[] array = {};
		int[] sortedArr = {};
		QuickSort.quicksort(array, 0, array.length-1);
		assertArrayEquals(sortedArr, array);
	}
	
	public void test_array_with_1_element() {
		int[] array = {2};
		int[] sortedArr = {2};
		QuickSort.quicksort(array, 0, array.length-1);
		assertArrayEquals(sortedArr, array);
	}
}

File: Quicksort_Python/QuickSortTest.py
import unittest
import QuickSort

class TestQuickSort(unittest.TestCase):

    def test_a_normal_array(self):
        array = [3, 2, -5, 1, 0]
        QuickSort.quicksort(array, 0, len(array)-1)
        self.assertEqual([-5, 0, 1, 2, 3], array)
    def test_an_empty_array(self):
        array = []
        QuickSort.quicksort(array, 0, len(array)-1)
        self.assertEqual([], array)
    def test_array_with_1_element(self):
        array = [1]
        QuickSort.quicksort(array, 0, len(array)-1)
        self.assertEqual([1], array)

if __name__ == '__main__':
    unittest.main()
